src/main/java/net/minecraftforge/fmp/block/BlockMultipartContainer.java

```diff
@@ -0,0 +1,561 @@
+    public static final IUnlistedProperty<List<PartState>> PROPERTY_MULTIPART_CONTAINER = new PropertyMultipartStates(
+            "multipart_container");
+
+    static final IProperty<?>[] PROPERTIES = new IProperty[]
+        { PROPERTY_TICKING };
+    static final IUnlistedProperty<?>[] UNLISTED_PROPERTIES = new IUnlistedProperty[]
+        { PROPERTY_MULTIPART_CONTAINER };
+
+    @Override
+    public IExtendedBlockState getExtendedState(IBlockState state, IBlockAccess world, BlockPos pos)
+    {
+        TileMultipartContainer tile = getMultipartTile(world, pos);
+        return ((IExtendedBlockState) state).withProperty(PROPERTY_MULTIPART_CONTAINER,
+                tile != null ? tile.getPartContainer().getExtendedStates(world, pos) : new ArrayList<PartState>());
+    }
+
+    @Override
+    public int getMetaFromState(IBlockState state)
+    {
+        return state.getValue(PROPERTY_TICKING) ? 0 : 1;
+    }
+
+    @Override
+    public IBlockState getStateFromMeta(int meta)
+    {
+        return getDefaultState().withProperty(PROPERTY_TICKING, meta == 0);
+    }
+
+    @Override
+    protected BlockStateContainer createBlockState()
+    {
+        return new ExtendedBlockState(this, PROPERTIES, UNLISTED_PROPERTIES);
+    }
+
+    @Override
+    public boolean canRenderInLayer(BlockRenderLayer layer)
+    {
+        return true;
+    }
+    
+    @Override
+    public boolean canRotateBlockAround(World world, BlockPos pos, EnumFacing axis)
+    {
+        TileMultipartContainer tile = getMultipartTile(world, pos);
+        if (tile != null)
+        {
+            for (IMultipart part : tile.getParts())
+            {
+                if (!part.canRotatePartAround(axis))
+                {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+    
+    @Override
+    public void rotateBlockAround(World world, BlockPos pos, EnumFacing axis)
+    {
+        TileMultipartContainer tile = getMultipartTile(world, pos);
+        if (tile != null)
+        {
+            for (IMultipart part : tile.getParts())
+            {
+                part.rotatePartAround(axis);
+            }
+        }
+    }
+    
+    @Override
+    public boolean canPlayerRotate(World world, BlockPos pos, EnumFacing axis, EntityPlayer player, RayTraceResult hit)
+    {
+        if (hit != null && hit.typeOfHit == Type.MULTIPART)
+        {
+            return hit.partHit.canPlayerRotate(axis, player, hit);
+        }
+        return false;
+    }
+    
+    @Override
+    public void rotateBlock(World world, BlockPos pos, EnumFacing axis, EntityPlayer player, RayTraceResult hit)
+    {
+        if (hit != null && hit.typeOfHit == Type.MULTIPART)
+        {
+            hit.partHit.rotatePart(axis, player, hit);
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    @SideOnly(Side.CLIENT)
+    public final void onDrawBlockHighlight(DrawBlockHighlightEvent event)
```

Comment:
Is there a reason for this handler to be in every block instance that could be created from here ?